src/main/java/com/atlassian/db/replica/api/DualConnection.java

```diff
@@ -28,18 +54,21 @@
     private final ReplicaConsistency consistency;
     private final DatabaseCall databaseCall;
     private final Set<String> readOnlyFunctions;
+    private final boolean compatibleWithPreviousVersion;
 
     private DualConnection(
         ConnectionProvider connectionProvider,
         ReplicaConsistency consistency,
         DatabaseCall databaseCall,
         StateListener stateListener,
-        Set<String> readOnlyFunctions
+        Set<String> readOnlyFunctions,
+        boolean compatibleWithPreviousVersion
     ) {
         this.connectionProvider = new ReplicaConnectionProvider(connectionProvider, consistency, stateListener);
         this.consistency = consistency;
         this.databaseCall = databaseCall;
         this.readOnlyFunctions = readOnlyFunctions;
+        this.compatibleWithPreviousVersion = compatibleWithPreviousVersion;
```

Comment:
So this is bound to the lifetime of a `DualConnection`. [`FeatureDecisions`](https://github.com/atlassian-labs/db-replica/pull/40#discussion_r581972605) could have been checked on every interaction.
It's a disadvantage for this solution, but doesn't look like a blocker.

Reply:
It's sooo tempting, I have to ask:
why not sth like `Map<String,Boolean>`with false return as a default. ? 
It solves problem of multiple introduced bugs, and slower development cycle because of dependency between them, but you will state that at the same time it introduce complexity.
Usually devs and good practices maintain complexity but use `feature power` to deliver faster.
WDYT?


aah shit, I know, I saw it somewhere:
https://github.com/atlassian-labs/db-replica/pull/40
Ok, I more fun of `Experiment API` plus I would couple it with semantic versions, and have an automated way to disable any Experiment in major releases.

Reply:
`Experiment AP`I introduces a broad API that will only grow and accumulate complexity. Removing API means major release, which is a pain for the library's consumers. Compatibility mode puts more constraints on the libraries maintenance but will not force us to choose between growing complexity and breaking compatibility.

Reply:
I think it's faster to go and try this solution here . It is simple to maintain and remove/replace if **it will impact our velocity with maintaining db-replica**, then we should go into `FeatureDecisions` and `Experiments` API